src/test/java/org/jenkins/plugins/lockableresources/LockStepTest.java

```diff
@@ -295,4 +296,105 @@ public void evaluate() throws Throwable {
+				story.j.waitForMessage("[resource1] is locked, waiting...", b2);
+
+				// Kill b1 hard.
+				b1.doKill();
+				story.j.waitForMessage("Hard kill!", b1);
+				story.j.waitForCompletion(b1);
+				story.j.assertBuildStatus(Result.ABORTED, b1);
+
+				// Now b2 is still sitting waiting for a lock. Create b3 and launch it to clear the lock.
+				WorkflowJob p3 = story.j.jenkins.createProject(WorkflowJob.class, "p3");
+				p3.setDefinition(new CpsFlowDefinition(
+						"lock('resource1') {\n"
+								+ "  semaphore 'wait-inside'\n"
+								+ "}"));
+				WorkflowRun b3 = p3.scheduleBuild2(0).waitForStart();
+
+				// Verify that b2 gets the lock.
+				story.j.waitForMessage("Lock acquired on [resource1]", b2);
+				SemaphoreStep.success("wait-inside/2", b2);
+				// Verify that b2 releases the lock and finishes successfully.
+				story.j.waitForMessage("Lock released on resource [resource1]", b2);
+				story.j.assertBuildStatusSuccess(story.j.waitForCompletion(b2));
+
+				// Now b3 should get the lock and do its thing.
+				story.j.waitForMessage("Lock acquired on [resource1]", b3);
+				SemaphoreStep.success("wait-inside/3", b3);
+				story.j.assertBuildStatusSuccess(story.j.waitForCompletion(b3));
+			}
+		});
+	}
+
+	// TODO: Figure out what to do about the IOException thrown during clean up, since we don't care about it. It's just
+	// a result of the first build being deleted and is nothing but noise here.
+	@Issue("JENKINS-36479")
+	@Test public void deleteRunningBuildNewBuildClearsLock() throws Exception {
+		story.addStep(new Statement() {
+			@Override public void evaluate() throws Throwable {
+				LockableResourcesManager.get().createResource("resource1");
+
+				WorkflowJob p1 = story.j.jenkins.createProject(WorkflowJob.class, "p");
+				p1.setDefinition(new CpsFlowDefinition("lock('resource1') { echo 'locked!'; semaphore 'wait-inside' }"));
+				WorkflowRun b1 = p1.scheduleBuild2(0).waitForStart();
+				story.j.waitForMessage("locked!", b1);
+				SemaphoreStep.waitForStart("wait-inside/1", b1);
+
+				WorkflowJob p2 = story.j.jenkins.createProject(WorkflowJob.class, "p2");
+				p2.setDefinition(new CpsFlowDefinition(
+						"lock('resource1') {\n"
+								+ "  semaphore 'wait-inside'\n"
+								+ "}"));
+				WorkflowRun b2 = p2.scheduleBuild2(0).waitForStart();
+
+				// Make sure that b2 is blocked on b1's lock.
+				story.j.waitForMessage("[resource1] is locked, waiting...", b2);
+
+				Thread.sleep(2000);
+				b1.delete();
+
+				// Now b2 is still sitting waiting for a lock. Create b3 and launch it to clear the lock.
+				WorkflowJob p3 = story.j.jenkins.createProject(WorkflowJob.class, "p3");
+				p3.setDefinition(new CpsFlowDefinition(
+						"lock('resource1') {\n"
+								+ "  semaphore 'wait-inside'\n"
+								+ "}"));
+				WorkflowRun b3 = p3.scheduleBuild2(0).waitForStart();
+
+				// Verify that b2 gets the lock.
+				story.j.waitForMessage("Lock acquired on [resource1]", b2);
```

Comment:
So, b2 is getting the lock because b3 release it before, right? Weird that a new build is required to unlock builds already waiting for the resource. Acceptable I guess, but weird.